application-mocca-calendar-ui/src/main/resources/MoccaCalendar/Code/Macro.xml

```diff
@@ -896,6 +899,71 @@ require(['jquery'], function($) {
     var calendarPopup = relatedTarget.data('calendarPopup');
     new XWiki.MoccaCalendar.MoccaCalendarPopup(calendarPopup.interactionParameters, calendarPopup.helper);
   });
+
+  // Trigger the upload manually when the button is clicked
+  $(document).on('click', '#importCalendarFileButton', function(event) {
+    var form = $('#importForm').get(0);
+    if (event) {
+      event.preventDefault();
+    }
+    startUploading(form);
+  });
+
+  // Start uploading this file, creating a new XHR object with the file data.
+  var startUploading = function (form) {
+    var formData = new FormData();
+    var input = form.down("input[type='file']");
+    var select = form.down("#importCalendarParent");
+    var file = input.files[0];
+    formData.append(input.name, file);
+    var urlQueryParameter = "?" + select.name + "=" + select.value;
+    formData.action = form.action + urlQueryParameter;
+
+    // Create XMLHttpRequest object and POST the data
+    var request = this.request = new XMLHttpRequest();
+    request.open('POST', formData.action);
+    request.send(formData);
+    checkImportJob(select.value);
+  }
+
+  const checkImportJob = function(selectedCalendar) {
+    let documentReference = XWiki.Model.resolve('MoccaCalendar.Code.ImportJobResource', XWiki.EntityType.DOCUMENT);
+    var targetURL = new XWiki.Document(documentReference).getURL('get');
+    var targetJobId = ["moccacalendar", "import", selectedCalendar]
+    var data = [];
+    data.push(
+      {name: 'outputSyntax', value: 'plain'},
+      {name: 'sheet', value: 'MoccaCalendar.Code.ImportJobResource'},
+      {name: 'action', value: 'jobStatus'},
+      {name: 'jobId', value: targetJobId.join('/')},
+      {name: 'data', value: 'jobStatus'},
+      {name: 'form_token', value: xwikiMeta.form_token}
+    );
+    var notification = new XWiki.widgets.Notification("$escapetool.javascript($services.localization.render('MoccaCalendar.import.notification.inprogress'))", 'inprogress');
```

Comment:
This is no longer the way for adding translations in js, check https://extensions.xwiki.org/xwiki/bin/view/Extension/Localization/#HFromJavaScript
For the others as well